build-logic/profiling/src/main/kotlin/gradlebuild.buildscan.gradle.kts

```diff
@@ -145,6 +146,10 @@ fun Project.extractCiData() {
 
 fun Project.isKillLeakingProcessesStep() = gradle.startParameter.taskNames.contains("killExistingProcessesStartedByGradle")
 
+fun Project.isPromotionBuild() = environmentVariable("BUILD_TYPE_ID").orNull?.let { buildTypeId ->
+    listOf("Promotion_Nightly", "Promotion_SnapshotFromQuickFeedback", "Promotion_PublishBranchSnapshotFromQuickFeedback").any { buildTypeId.contains(it) }
```

Comment:
I was wondering if we could use that we run the `promotionBuild` task instead of looking at the environment? WDYT?

Reply:
I realized there's already a `isPromotionBuild`, so let's just reuse it.